src/two_stage_method.jl

```diff
@@ -102,23 +72,27 @@ get_chunksize(cs) = cs
 get_chunksize(cs::Type{Val{CS}}) where {CS} = CS
 
 function two_stage_method(prob::DiffEqBase.DEProblem, tpoints, data;
-                          kernel = EpanechnikovKernel(),
+                          kernel::Union{CollocationKernel, Symbol} = EpanechnikovKernel(),
                           loss_func = L2Loss, mpg_autodiff = false,
                           verbose = false, verbose_steps = 100,
                           autodiff_chunk = length(prob.p))
-    f = prob.f
-    kernel_function = decide_kernel(kernel)
-    estimated_derivative, estimated_solution = construct_estimated_solution_and_derivative!(data,
-                                                                                            kernel_function,
-                                                                                            tpoints)
+    if kernel isa Symbol
+        @warn "Passing kernels as Symbols will be deprecated"
```

Comment:
You can use @deprecate here

Reply:
I wanted to do it like that at first, but I ran into issues because Julia cannot dispatch on a keyword argument, i.e. it is impossible to have two methods, one with kernel restricted to `Symbol` types and one with kernel restricted to `CollocationKernel`, and then `@deprecate` the `Symbol` method.

Reply:
I don't think you can `@depreacate` here, it needs this special handling. But I also don't think we should be merging deprecations to this repo: it's somewhat in freeze mode because its interface does not extend well to more cases. In that case, major dependency changes and deprecations for an old repo not getting any major benefits is not a great feel. I think this PR should just hold until there's a good reason found for it.

Reply:
Yeah without kwarg handling it's not obvious, I think the way to do it with `@deprecate` would be 
(though it's not relevant anymore since we won't go ahead with the PR, but it might be useful later)

```
my_func(args...; kwarg_to_deprecate::Union{T1,T2}, kwargs...) = my_func(args..., kwarg_to_deprecate; kwargs...)

my_func(args..., kwarg_to_deprecate::T1; kwargs....) = do something...

@deprecate my_func(args..., kwarg_to_deprecate::T2; kwargs...) = do something....
```